File: app/src/main/java/ir/beigirad/zeroapplication/network/APIProvider.java
package ir.beigirad.zeroapplication.network;

import com.facebook.stetho.okhttp3.StethoInterceptor;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Created by farhad-mbp on 7/17/17.
 */

public class APIProvider {

    Retrofit mRetrofit;
    APIService mApiService;

    public APIProvider() {
        OkHttpClient.Builder client = new OkHttpClient.Builder();
        client.addNetworkInterceptor(new Interceptor() {
            @Override
            public Response intercept(Chain chain) throws IOException {
                Request original = chain.request();
                Request.Builder builder = original.newBuilder();

                builder.addHeader("user_id", "fucking_user_id");
                builder.addHeader("token", "fucking_user_token");

                return chain.proceed(builder.build());
            }
        });

        client.addNetworkInterceptor(new StethoInterceptor());

        mRetrofit = new Retrofit.Builder()
                .client(client.build())
                .baseUrl("http://beigirad.ir/")
                .addConverterFactory(GsonConverterFactory.create())
                .build();
        mApiService = mRetrofit.create(APIService.class);

    }

    public APIService getApiService() {
        return mApiService;
    }

}

File: app/src/main/java/ir/beigirad/zeroapplication/network/APIProvider.kt
package ir.beigirad.zeroapplication.network

import com.facebook.stetho.okhttp3.StethoInterceptor
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

/**
 * Created by farhad-mbp on 11/4/17.
 */
public class APIProvider {
    private val BASE_URL = "http://46.32.17.139:9080"

    private var okHttpClient: OkHttpClient

    private var mRetrifit: Retrofit? = null
    private var mAPIService: APIService? = null

    init {
        var builder = OkHttpClient.Builder()
        builder.addNetworkInterceptor(StethoInterceptor())

        builder.addInterceptor { chain ->
            val build = chain?.request()?.newBuilder()
//                        ?.addHeader("Authorization", "")
                    ?.build()

            chain!!.proceed(build)
        }
        okHttpClient = builder.build()
        mRetrifit = Retrofit.Builder()
                .client(builder.build())
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build()

        mAPIService = mRetrifit!!.create(APIService::class.java)
    }


    fun getAPIService(): APIService {
        return mAPIService!!
    }

    fun getRetrofit(): Retrofit {
        return mRetrifit!!
    }

    fun cancellAll() {
        okHttpClient.dispatcher().cancelAll()
    }

}